wp_api/src/lib.rs

```diff
@@ -73,6 +75,8 @@ pub enum WpApiParamOrder {
     Desc,
 }
 
+impl_as_query_value_from_as_str!(WpApiParamOrder);
```

Comment:
Is it possible to implement a `#[derive(AsQueryValue)]` to do the same? This is not a request for changes though.

Reply:
It's, but the implementation would be more complicated because if we are going to have a single derive such as `AsQueryValue`, then we'd need to have options for it, such as `from=as_str`, `from=to_string` etc. Handling that is more complicated, so it add maintenance burden. We could add different derives such as `AsQueryValueFromAsStr`, `AsQueryValueFromToString`, but that's not any better unfortunately.

We are trying to implement derive macros when there is significant benefit to them compared to other solutions. This is not such a case, because the simple `macro_rules!` does a good job and it is easier to understand.

Fwiw, for a personal project I'd probably go with the derive macro as I am used to working with them. However, I don't think it's the right choice for this project - at least not at this time.

Reply:
Thanks for the explanation!